src/null-safety/index.md

```diff
@@ -1,39 +1,247 @@
+**add `!`** to tell the Dart analyzer that the value isn’t null:
+
+```dart
+int? a = 2;
+int b = a!; // The ! temporarily changes `a` from int? to int.
+```
+
+If you know that a non-nullable value will be
+initialized to a non-null value before it's used,
+but the Dart analyzer thinks otherwise,
+**add the `late` keyword** to the type:
+
+```dart
+late int aRealInt;
+```
+
+{% comment %}
+{{ site.alert.tip }}
+  Avoid using `late final` unless you really need it.
+  For example, don't use `late final` in functions,
+  and in constructors prefer using an [initializer list][].
+{{ site.alert.end }}
+
+[initializer list]: /guides/language/language-tour#initializer-list
+{% endcomment %}
+
+
+## Understanding list, set, and map types
+
+Lists, sets, and maps are commonly used collection types in Dart programs,
+so you need to know how they interact with null safety.
+Here are some examples of how Dart code uses these collection types:
+
+* Flutter layout widgets such as [`Column`][]
+  often have a `children` property that’s a
+  `List` of `Widget` objects.
+* The [Veggie Seasons][] sample uses a `Set` of `VeggieCategory` to
+  store a user’s food preferences.
+* The [GitHub Dataviz][] sample has a `fromJson()`method that
+  creates an object from JSON data that’s supplied in a
+  `Map<String, dynamic>`.
+
+[`Column`]: https://api.flutter.dev/flutter/widgets/Column-class.html
+[Veggie Seasons]: https://github.com/flutter/samples/tree/master/veggieseasons
+[GitHub Dataviz]: https://github.com/flutter/samples/tree/master/web/github_dataviz
+
+
+### List and set types {#list-and-set-types}
+
+When you’re declaring the type of a list or set, think about what can be null. The following table shows the possibilities for a list of strings
+if you opt into null safety.
+
+{% assign yes = '<b>Yes</b>' %}
+
+|------------------+----------------------------------+--------------------------+------------|
+|Type              | Can an item (string)<br>be null? | Can the list<br>be null? | Description|
+|------------------|----------------------------------|--------------------------|------------|
+| `List<String>`   | No      | No      | A non-null list that contains<br> non-null strings       |
+| `List<String>?`  | No      | {{yes}} | A non-null list that contains<br> strings that **might be null** |
+| `List<String?>`  | {{yes}} | No      | A list that **might be null** and that<br> contains non-null strings |
+| `List<String?>?` | {{yes}} | {{yes}} | A list that **might be null** and that<br> contains strings that **might be null**<br> (same as `List<String>` if you <br>don't opt into null safety) |
+{:.table .table-striped}
+
+When literals initialize a list or set,
+then instead of seeing a type like in the table above,
+you typically see a type annotation on the literal.
+For example, here’s what a `List<String?>` (called `nameList`) and
+a `Set<String?>` (called `nameSet`) might look like:
```

Comment:
-> here's what variables containing a ....

The `nameList` variable is not a `List<String?>`. It's (current) value *is* a `List<String?`>, and its type is `List<String?>`. When *reading* a variable, it's sometimes OK to talk about it as if the variable is its value, but here we are declaring the variables, not reading them.